### PR TITLE
revamp build system - work in progress

### PR DESCRIPTION
**Pull request by jgravois**
i followed the pattern in Esri/esri-leaflet-geocoder#122 to a *T*, but my new builds don't put anything in the `L.esri.Renderers` namespace.

not sure what i could be missing
  * maybe this plugin needs something unique since it extends the functionality of existing `L.esri` classes?
  * do i need to update [leaflet.shapeMarkers](https://github.com/Esri/leaflet.shapeMarkers) concurrently? (i wouldn't think so)

cc/ @patrickarlt

### CODE REVIEW

**Comment by jgravois** [comment]
looking at `esri-leaflet-renderers-debug.js`, Leaflet itself isn't being mapped correctly, but i'm still scratching my head as to why...
```js
L$1 = 'default' in L$1 ? L$1['default'] : L$1;
// instead of
L = 'default' in L ? L['default'] : L;
```

**Comment by jgravois** [comment]
@kneemer confirmed that your tweaks fixed the new build.
i saw the same behavior you reported [here](https://github.com/Esri/esri-leaflet-renderers/issues/105#issuecomment-194686736) the first time i launched `debug/sample.html`

**Comment by kneemer** [comment]
@jgravois let me know if you can get some steps for reproduction.  I haven't been able to yet.

**Comment by jgravois** [comment]
@kneemer i can't repro dependably either, even when clearing the browser cache.  perhaps tracking down some more complex geometry would make the problem appear more often.

**Comment by kneemer** [comment]
Did you happen to have your console open when seeing the issue, I didn't. :disappointed:   If so, were there any errors?

**Comment by jgravois** [comment]
> Did you happen to have your console open when seeing the issue?

yup.  i can still reproduce intermittently.  no errors appear in the console.

**Comment by kneemer** [comment]
Well I still don't have repo steps but I was able to narrow it down to the following function call sequence:
1. onAdd gets called
2. createLayers (in esri-leaflet) 
3. metadata

When everything works as expected, the second step only happens after the metadata call.

**Comment by kneemer** [comment]
Able to reproduce in code by setting a timeout around https://github.com/Esri/esri-leaflet/blob/master/src/Services/Service.js#L35 to simulate a slow call to metadata.  Working on a fix now.

**Comment by kneemer** [comment]
@jgravois after successfully reproducing the issue, reverting #107 fixes the problem.  Will need a different solution for the race condition.  Maybe we can create a special branch for PatrickKing until we can get a better solution and revert the change?

**Comment by jgravois** [comment]
you're the one figuring everything out, so however you want to approach the fix is fine by me.

File: src/EsriLeafletRenderers.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-export var VERSION = '2.0.1';
+export { version as VERSION } from '../package.json';
 
-export { Renderer, renderer } from './Renderers/Renderer';
```
Comment (kneemer) [inline src/EsriLeafletRenderers.js]: hi

File: src/EsriLeafletRenderers.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 export { version as VERSION } from '../package.json';
 
-export { Renderer, renderer } from './Renderers/Renderer';
+export { Renderer } from './Renderers/Renderer';
 export { SimpleRenderer, simpleRenderer } from './Renderers/SimpleRenderer';
```
Comment (kneemer) [inline src/EsriLeafletRenderers.js]: hello there

**Comment by jgravois** [comment]
yikes.  thanks for the reminder @kneemer.  i have reverted your commit from #105 and held off on bumping the project version.  i'll review the other open PR and do a quick sanity check to make sure nothing else is screwed up before tagging another release.

**Pull request merged**